frameworks/RandomForest/exec.py

```diff
@@ -16,29 +18,69 @@
 log = logging.getLogger(os.path.basename(__file__))
 
 
+def extrapolate_with_worst_case(values: List[float], n: int = 5) -> float:
+    """ Extrapolate the next value for `values`, based on the last `n` samples. """
+    n = min(len(values), n)
+    return values[-1] + max(v_next - v_prev for v_prev, v_next in zip(values[-n:], values[-n + 1:]))
+
+
 def run(dataset, config):
     log.info(f"\n**** Random Forest [sklearn v{sklearn.__version__}] ****\n")
 
     is_classification = config.type == 'classification'
+    this_process = psutil.Process(os.getpid())
 
     encode = config.framework_params.get('_encode', True)
     X_train, X_test = dataset.train.X, dataset.test.X
     y_train, y_test = dataset.train.y, dataset.test.y
 
-    training_params = {k: v for k, v in config.framework_params.items() if not k.startswith('_')}
+    training_params = {
+        k: v for k, v in config.framework_params.items()
+        if not (k.startswith('_') or k == "n_estimators")
+    }
     n_jobs = config.framework_params.get('_n_jobs', config.cores)  # useful to disable multicore, regardless of the dataset config
+    step_size = config.framework_params.get('_step_size', 10)
+    final_forest_size = config.framework_params.get('n_estimators', 2000)
+
+    # Default margins are conservative, because robustness is paramount for a baseline.
+    time_margin = config.framework_params.get('_time_margin', 0.9)
+    memory_margin = config.framework_params.get('_memory_margin', 0.9)
 
     log.info("Running RandomForest with a maximum time of {}s on {} cores.".format(config.max_runtime_seconds, n_jobs))
-    log.warning("We completely ignore the requirement to stay within the time limit.")
     log.warning("We completely ignore the advice to optimize towards metric: {}.".format(config.metric))
 
     estimator = RandomForestClassifier if is_classification else RandomForestRegressor
     rf = estimator(n_jobs=n_jobs,
                    random_state=config.seed,
+                   n_estimators=step_size,
+                   warm_start=True,
```

Comment:
If you warm-start with the same random seed, won't you just add the same trees over and over again? From looking at the code, it seems you would add the same 10 trees over and over again to the ensemble?

Reply:
Sharp eye. I actually had to test this myself since I could not find clarification on this in the documentation:
```
from sklearn.datasets import make_classification
from sklearn.ensemble import RandomForestClassifier

x, y = make_classification(random_state=0)
rf = RandomForestClassifier(n_estimators=1, warm_start=True, random_state=0)

rf.fit(x, y)
rf.n_estimators += 1
rf.fit(x, y)

print(rf.estimators_)
```
> [DecisionTreeClassifier(max_features='auto', random_state=209652396), DecisionTreeClassifier(max_features='auto', random_state=398764591)]

Repeating this script more than once gives identical output (which in turn is identical to just fitting the forest with `n_estimators=2` in a single fit). You can further look at the trees to see that they are indeed distinct:
```
from sklearn.tree import export_text
for tree in rf.estimators_:
    print(export_text(tree))
```
So it seems that the usage here is correct and does not require any modification.

Reply:
OK, so warm-starting overrules the random state? 
Or does it internally start another RNG using random_state=0? You you get seeds 209652396 and 398764591 if you run it again?
That is probably good but something that should be documented :).

Reply:
A `random_state` results in the same forest (with unique trees) whether or not `warm_start` is set, even when `warm_start=True` and the forest is created through multiple `fit` calls. This of course contradicts that every `fit` call will yield identical results.